### PR TITLE
♻️ Refactor and add retries

### PR DESCRIPTION
**Pull request by asim-shrestha**

### CODE REVIEW

File: next/src/components/console/ChatMessage.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ const getMessagePrefix = (message: Message) => {
   } else if (getTaskStatus(message) === TASK_STATUS_COMPLETED) {
     return `Executing: ${message.value}`;
   } else if (getTaskStatus(message) === TASK_STATUS_FINAL) {
-    return `Finished: ${message.value}`;
+    return `Finished:`;
```
Comment (awtkns) [inline next/src/components/console/ChatMessage.tsx:66]:
```suggestion
    return `Finished.`;
```

File: next/src/services/agent/autonomous-agent.ts [hunk 2]
```diff
@@ -33,131 +35,219 @@ class AutonomousAgent {
...
+            await new Promise((r) => setTimeout(r, RETRY_TIMEOUT));
+          }
+
+          return shouldContinue;
+        }
+      );
+      useAgentStore.getState().setIsAgentThinking(false);
+
+      this.workLog.shift();
+      if (this.isRunning) {
```
Comment (awtkns) [inline next/src/services/agent/autonomous-agent.ts:83]: invert this condition

File: next/src/services/agent/autonomous-agent.ts [hunk 3]
```diff
@@ -33,131 +35,219 @@ class AutonomousAgent {
...
+
+    onError = (e: unknown): boolean => {
+      this.parent.messageService.sendErrorMessage(e);
+      return false;
+    };
+
+    next = () => undefined;
+  };
+
+  analyzeTaskWork = class AnalyzeTaskWork implements AgentWork {
```
Comment (awtkns) [inline next/src/services/agent/autonomous-agent.ts:135]: weird but ok

File: next/src/services/agent/autonomous-agent.ts [hunk 4]
```diff
@@ -33,131 +35,219 @@ class AutonomousAgent {
...
+
+    run = async () => {
+      this.parent.messageService.startTaskMessage(this.task);
+      this.task = this.parent.model.updateTaskStatus(this.task, "executing");
+      this.analysis = await this.parent.$api.analyzeTask(this.task.value);
+    };
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    conclude = async () => {
+      if (!this.analysis) {
```
Comment (awtkns) [inline next/src/services/agent/autonomous-agent.ts:148]: invert this for clairity

File: next/src/services/agent/autonomous-agent.ts [hunk 5]
```diff
@@ -33,131 +35,219 @@ class AutonomousAgent {
...
+        this.parent.messageService.skipTaskMessage(this.task);
+      } else {
+        this.parent.messageService.sendAnalysisMessage(this.analysis);
+      }
+    };
+
+    next = () => {
+      if (!this.analysis) return undefined;
+      return new this.parent.executeTaskWork(this.parent, this.task, this.analysis);
+    };
```
Comment (awtkns) [inline next/src/services/agent/autonomous-agent.ts:158]:
```suggestion
    next = () => this.analysis ? new this.parent.executeTaskWork(this.parent, this.task, this.analysis) : undefined;
```

File: next/src/services/agent/message-service.ts [hunk 6]
```diff
@@ -82,10 +87,12 @@ export class MessageService {
...
   sendErrorMessage(e: unknown) {
     let message = "An unknown error occurred. Please try again later.";
-
     if (typeof e == "string") message = e;
-    else if (axios.isAxiosError(e)) {
+    else if (axios.isAxiosError(e) && e.message == "Network Error") {
+      message = "Error attempting to connect to the server.";
+    } else if (axios.isAxiosError(e)) {
       const data = (e.response?.data as object) || {};
+      console.log(data);
```
Comment (awtkns) [inline next/src/services/agent/message-service.ts:95]:
```suggestion
```

File: next/src/components/console/ChatMessage.tsx (hunk 1 above)
Comment (asim-shrestha) [reply next/src/components/console/ChatMessage.tsx:66]: Some jank with task coupling. It will display the task value after this 🙃

**Pull request merged**